src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -403,8 +403,14 @@ private static bool TryProcessTestDataSourceTests(UnitTestElement test, TestMeth
     private static bool ProcessTestDataSourceTests(UnitTestElement test, ReflectionTestMethodInfo methodInfo, IEnumerable<ITestDataSource> testDataSources,
```

Comment:
A different possible fix is to change:

https://github.com/microsoft/testfx/blob/04f901ceb6e7069a65c2734ab49081ba1fd5c07c/src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs#L288

to:

```csharp
if (test.TestMethod.DataType != DynamicDataType.ITestDataSource)
```

Reply:
I think this might be a better fix to make, together with a comment that DataSourceAttribute tests connect to external systems like databases, and we generally want to postpone that until run.

And then renaming the ProcessTestDataSourceTests to ProcessITestDataSourceTests, and adding a comment that this does no apply to DataSourceAttribute tests. 

Is this code path taken only when TestDataSourceDiscoveryOption.DuringDiscovery is set? It might be also worth to mention that in the comment.